### ISSUE TITLE
Can't extract large files?

### ISSUE THREAD

**Issue by alechenninger** [brendan-duncan/archive#13]
Try decompressing all of https://storage.googleapis.com/dart-archive/channels/dev/release/latest/dartium/dartium-linux-x64-release.zip

For me this gets caught in a [seemingly infinite] loop at the chrome binary, with 1g (99%) of old gen full of allocated lists.

**Comment by brendan-duncan** [comment]
I'll look into it.  If it's a memory issue, then it may require this solution, which I plan on addressing soon: https://github.com/brendan-duncan/archive/issues/12

**Comment by brendan-duncan** [comment]
The slowness decompressing that chrome file from the zip (or any significantly large file) was due to not pre-allocating the output buffer, causing it to reallocate and copy the buffer every 32k, which has a significant hit on performance.  I fixed it to pre-allocate the output buffer to the known uncompressed size of the file.  The zip file extracted correctly for me in a reasonable amount of time now.  I'll push a pub update with this fix. Thanks for pointing out the problem.
